src/devices/bus/isa/ibm_speech.cpp

```diff
@@ -0,0 +1,497 @@
+
+    Selects the audio source from the four possible sources. Only one source
+    can be used at a time.
+******************************************************************************/
+void isa8_ibm_speech_device::channel_mux_w(uint8_t data)
+{
+	switch(data & 3)
+	{
+		case 0:
+			LOGMASKED(LOG_MUX, "Channel MUX: source now LPC\n");
+			m_lpc->set_output_gain(0, 1.0);
+			m_cvsd->set_output_gain(0, 0.0);
+			m_beeper_gate = 0;
+			break;
+		case 1:
+			LOGMASKED(LOG_MUX, "Channel MUX: source now CVSD\n");
+			m_lpc->set_output_gain(0, 0.0);
+			m_cvsd->set_output_gain(0, 1.0);
+			m_beeper_gate = 0;
+			break;
+		case 2:
+			LOGMASKED(LOG_MUX, "Channel MUX: source now 8254\n");
+			m_lpc->set_output_gain(0, 0.0);
+			m_cvsd->set_output_gain(0, 0.0);
+			m_beeper_gate = 1;
+			break;
+		case 3:
+			LOGMASKED(LOG_MUX, "Channel MUX: source now Test Signal\n");
+			// TODO: How is this generated?
+			break;
+	}
+}
+
+WRITE_LINE_MEMBER(isa8_ibm_speech_device::lpc_interrupt_w)
+{
+	// Active low.
+
+	// Pulse IRQ7 on falling edge.
+	if(m_lpc_interrupt && !state)
+	{
+		LOGMASKED(LOG_IRQ, "pulsing IRQ7\n");
+		m_isa->irq7_w(ASSERT_LINE);         // Raise IRQ...
+		m_timers->write_gate2(ASSERT_LINE); // ...CH2 gate open, triggering CH2 one-shot.
+	}
+	else if(!m_lpc_interrupt && !state)
+	{
+		LOGMASKED(LOG_IRQ, "IRQ7 already raised, go away\n");
+	}
+	else if(state)
+	{
+		LOGMASKED(LOG_IRQ, "LPC INT pin going high, CH2 gate closed\n");
+		m_isa->irq7_w(CLEAR_LINE);
+		m_timers->write_gate2(CLEAR_LINE);
+	}
+
+	m_lpc_interrupt = state;
+}
+
+/******************************************************************************
+    CVSD Clock - PIT CH0
+
+    Sets the speed of the CVSD bitstream.
+
+    Non-inverted to:
+        - MC3418 CLK
+    Inverted to:
+        - PIT CLK1
+        - ZM12 serial clock
+        - PPI PORTC.3
+******************************************************************************/
+WRITE_LINE_MEMBER(isa8_ibm_speech_device::cvsd_clock_w)
+{
+	m_cvsd_clock = state;
+
+	LOGMASKED(LOG_CVSD, "cvsd_clock_w %d\n", state);
+
+	// PIT CH0 is the CVSD CLOCK signal.
+	m_cvsd->clock_w(state);         // Straight through to the CVSD CLK pin.
+
+	// Through an inverter and to:
+	//m_timers->write_clk1(state ? 0 : 1);  // PCjr: PIT CLK1
+	cvsd_shiftreg_clk_w(state ? 0 : 1);     // 74859 clock pin
+											// PPI PC.3 (handled over there)
+}
+
+/******************************************************************************
+    CVSD Frame - PIT CH1
+
+    Defined as (CVSD Clock / 8).
+    This signal goes low for one clock every 8 CVSD bit clocks, which specifies
+    how often a new byte gets latched into the CVSD shift register.
+******************************************************************************/
+WRITE_LINE_MEMBER(isa8_ibm_speech_device::cvsd_frame_w)
+{
+	// PIT CH1 is the CVSD FRAME signal.
+	if(state)   LOGMASKED(LOG_CVSD, "*** CVSD frame going high\n");
+	else        LOGMASKED(LOG_CVSD, "*** CVSD frame going low\n");
```

Comment:
Can be simplified with:
`LOGMASKED(LOG_CVSD, "*** CVSD frame going %s\n", state ? "high" : "low");`